### PR TITLE
[Android] Add data version filter support in IM

### LINKED ISSUE #25903: [Android controller]Expose data version filter in read and subscribe Java/JNI API

### PR DESCRIPTION
**Pull request by yunhanw-google**
Add data version filter support in Android controller's read/subscribe API.

fixes: https://github.com/project-chip/connectedhomeip/issues/25903

### CODE REVIEW

**Review by joonhaengHeo — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

File: src/app/ReadClient.cpp [hunk 1]
```diff
@@ -693,7 +693,7 @@ CHIP_ERROR ReadClient::ProcessAttributeReportIBs(TLV::TLVReader & aAttributeRepo
             ReturnErrorOnFailure(data.GetDataVersion(&version));
             attributePath.mDataVersion.SetValue(version);
 
-            if (mReadPrepareParams.mpDataVersionFilterList != nullptr)
+            if (mReadPrepareParams.mpDataVersionFilterList != nullptr && mReadPrepareParams.mDataVersionFilterListSize != 0)
```
Comment (bzbarsky-apple) [inline src/app/ReadClient.cpp:696]: @yunhanw-google Why was this change needed?

Comment (yunhanw-google) [reply src/app/ReadClient.cpp:696]: thanks, it is not needed, but i think it is better to have?

Comment (bzbarsky-apple) [reply src/app/ReadClient.cpp:696]: Why?
